rosbag2_py/test/common.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import rosbag2_py
+
+
+def get_rosbag_options(path, serialization_format='cdr'):
```

Comment:
``metadata.yaml`` and sqlite table for topics lists a serialization_format per topic. Does this mean we need one reader per serialization_format or is the some way to achieve this with one reader?

Is this a real-world scenario to have different serialization formats within one rosbag2?

Reply:
IIUC, the reader will convert messages with different serialization format to the format provided by the options: https://github.com/ros2/rosbag2/blob/160af0fc73261bc7d7105f4f85cfd034bed15f44/rosbag2_cpp/include/rosbag2_cpp/reader.hpp#L81-L84

Reply:
@jacobperron I was/am thrown of by the converter_options.input_serialization_format set below.

Reply:
I think `input_serialization_format` is used by the writer; the writer will convert messages to this format when storing them.

This made me realize that in C++ there's an alternative of **not** providing storage and converter options for both readers and writers. If no `input_serialization_format` is provided, then it will use the format of the incoming messages. Similarly, if no `output_serialization_format` is provided, then messages are read in whatever format they were stored as.

For the both the readers and writers, if we want to support *not* specifying the serialization format in Python, then we should expose these overloaded `open` functions:

https://github.com/ros2/rosbag2/blob/160af0fc73261bc7d7105f4f85cfd034bed15f44/rosbag2_cpp/include/rosbag2_cpp/reader.hpp#L74

https://github.com/ros2/rosbag2/blob/160af0fc73261bc7d7105f4f85cfd034bed15f44/rosbag2_cpp/include/rosbag2_cpp/writer.hpp#L72

If this sounds like a feature we want in the Python API, it might be better to do it in a follow-up PR to avoid adding more on to this one.